File: Ugly-Number/UN.py
def isUgly(num):
    while(1):
        if num == 1 or num == 2 or num == 3 or num == 5:
            return True
        elif num == 0:
            return False
        elif num % 2 == 0:
            num = num / 2
            continue
        elif num % 3 == 0:
            num = num / 3
            continue
        elif num % 5 == 0:
            num = num / 5
            continue
        else:
            return False

File: Ugly-Number/UN.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

var args []string = os.Args[1:]

func IsUgly(num int) bool {
	var re bool
	switch {
	case IfInOrNot(num, [4]int{1, 2, 3, 5}) == true:
		re = true
	case num == 0:
		re = false
	case num%3 == 0:
		re = IsUgly(num / 3)
	case num%5 == 0:
		re = IsUgly(num / 5)
	case num%2 == 0:
		re = IsUgly(num / 2)
	}
	return re
}

func IfInOrNot(num int, pool [4]int) bool {
	var re bool
	for _, i := range pool {
		if i == num {
			re = true
			break
		} else {
			re = false
		}
	}
	return re
}

func main() {
	num, _ := strconv.Atoi(args[0])
	fmt.Print(IsUgly(num))
}
